portfolio-website/script.js

```diff
@@ -1,23 +1,79 @@
+// navBar
 const navBarList = [
-    "Ingredients", "Steps to Make", "Links to Other Wonton Soup Recipes", "About Me"
+    "About Me", "Projects", "Contact"
 ]
 
 for (let i = 0; i < navBarList.length; i++) {
     const label = document.createElement('a');
     label.innerText = navBarList[i];
     label.addEventListener('mouseover', function() {
-        label.style.color = "white";
+        label.style.color = "black";
     })
     label.addEventListener('mouseout', function() {
-        label.style.color = "black";
+        label.style.color = "white";
     })
     label.addEventListener('click', function() {
         label.setAttribute('href', `#${navBarList[i].split(" ").join("")}`);
-        if (i === 3) {
-            label.setAttribute('href', "./aboutme.html");
+        if (i === 2) {
+            label.setAttribute('href', "./contactForm.html")
         }
         label.style.textDecoration = "none";
     })
 
     document.getElementById("navBar").appendChild(label);
-}
\ No newline at end of file
+}
+// end navBar
+
+// typewriter functionality
+const span1 = document.createElement('span');
+span1.textContent = "Welcome to";
+const img = document.createElement('img');
+img.setAttribute('src', "https://miro.medium.com/max/875/1*b2ZpVczibdIDlUbuN6S2ZA.png");
+img.setAttribute('id', "pfp");
+const span2 = document.createElement('span');
+span2.textContent = " my website!";
+document.querySelector("#first").append(span1);
+document.querySelector("#last").append(span2);
+let quoteArray = ["Welcome to"];
+let quoteArray2 = ["             my website!"];
+let textPosition = 0; 
+let textPosition2 = 0; 
+let speed = 100;
+typewriter1 = () => {
+    span1.innerHTML = quoteArray[0].substring(0, textPosition);
+  
+    if(textPosition++ != quoteArray[0].length) {
+        setTimeout(typewriter1, speed);
+    }
+}
+typewriter2 = () => {
```

Comment:
typewriter1 repeats typewriter2, could definitely DRY / don't repeat yourself and use closures to create two. 

```
const createTyperwriter = (element, quote, position) => {
      const typewriter = () => {
       element.innerHtml = quote[0].substring(0, position);
       if (textPosition++ !== quote[0].length){
             setTimeout(typewriter, SPEED);
       }
       return typewriter;
   }
}

const prefixQuoteTypewriter = createTyperwriter(span1, prefixQuote, 0); 
const suffixQuoteTypewriter = createTyperwriter(span2, suffixQuote, 0);

window.addEventListener("load", prefixQuoteTypewriter);
window.addEventListener("load", suffixQuoteTypewriter);
```

Reply:
Before I saw this comment, I had tried to do it like this!
`function createTypewriter(location, quote, SPEED) {
    const txt = document.createElement('span');
    document.querySelector(`#${location}`).append(txt);
    let txtPosition = 0;
    return function typewriter() {
        txt.innerHTML = quote[0].substring(0, txtPosition);

        if (txtPosition++ != quote[0].length) {
            setTimeout(typewriter, SPEED)
        }
    }
}

const typeFirst = createTypewriter("first", ["Welcome to"], 100);
const typeSecond = createTypewriter("last", [`           my website!`], 100);

window.addEventListener("load", typeFirst);
window.addEventListener("load", typeSecond);`

Reply:
this looks really great! nice! 😎